### PR TITLE
Adds routeTo method for HATEOAS support

### PR DESCRIPTION
**Pull request by jamestharpe**
Addresses #225 . 

Sorry, I wasn't sure how to organize tests for this. Happy to update it w/ some guidance.

This makes it easy to add HATEOAS style links to response models.

Example: 

```typescript
@Get('{parentId}')
async getParent(parentId: string): Promise<Parent> {
  const result = {
    id: parentId,
    name: "Daddy",
    links: [
      { 
        rel: 'children',
        href: routeTo(ChildrenController.prototype.getChildren, {
            parentId: parentId
          })
      }
    ]
  }
}
```

### CODE REVIEW

**Comment by endor** [comment]
Maybe you could add a file to `tests/unit/swagger/common` and add a few tests cases for the existing `GetTestController` with valid and invalid uses of `routeTo`.

File: src/decorators/methods.ts [hunk 1]
```diff
@@ -1,19 +1,56 @@
...
+    for (const argProp of argProps) {
+      pathValue = pathValue.replace('{' + argProp + '}', args[argProp]);
+    }
+  }
+  return (
+    '/' + baseRoute + '/' + encodeURI(pathValue)
+  );
+}
+
+// const saveRoute =
```
Comment (endor) [inline src/decorators/methods.ts]: is this intentional?

**Comment by jamestharpe** [comment]
Thanks for the feedback, I'll make some updates!

File: src/decorators/methods.ts (hunk 1 above)
Comment (jamestharpe) [reply src/decorators/methods.ts]: Nope! I'll update, thanks!

**Comment by dgreene1** [comment]
@jamestharpe if you want to update this with tests, we’d love to introduce hateos support. There are a lot of recent PRs that provide examples. Additionally we’d need to see some tests that verify that the Schema output has the proper links information as described in openAPI: https://swagger.io/docs/specification/links/

In the meantime I’m going to close the PR since it’s rather old. If you’d like to reopen it, please feel free to. :)

**Pull request closed without merging**